container-engine-lib/lib/backend_impls/kubernetes/kubernetes_manager/kubernetes_manager.go

```diff
@@ -1748,6 +2070,140 @@ func (manager *KubernetesManager) RunExecCommandWithStreamedOutput(
 	return execOutputChan, finalExecResultChan, nil
 }
 
+// RemoveDirPathFromNode removes the contents and path to [dirPathToRemove] by creating a pod in [namespace] with privileged access to the [nodeName]'s filesystem
+// The host filesystem is mounted onto the pod as a volume and then a rm -rf is run at the location on the pod where [dirPathToRemove] is mounted
+func (manager *KubernetesManager) RemoveDirPathFromNode(ctx context.Context, namespace string, nodeName string, dirPathToRemove string) error {
+	// rm the fluent bit directory from the node using a privileged pod
+	removeContainerName := "remove-dir-container"
+	// pod needs to be privileged to access host filesystem
+	isPrivileged := true
+	hasHostPidAccess := true
+	hasHostNetworkAccess := true
+	removeDataDirPodName := "remove-dir-pod"
+	hostVolumeName := "host"
+	hostMountPath := "/host"
+	removeDataDirPod, err := manager.CreatePod(ctx,
+		namespace,
+		removeDataDirPodName,
+		nil,
+		nil,
+		nil,
+		[]apiv1.Container{
+			{
+				Name:  removeContainerName,
+				Image: "busybox",
+				Command: []string{
+					"sh",
+					"-c",
+					"sleep 10000000s",
+				},
+				Args:       nil,
+				WorkingDir: "",
+				Ports:      nil,
+				EnvFrom:    nil,
+				Env:        nil,
+				Resources: apiv1.ResourceRequirements{
+					Limits:   nil,
+					Requests: nil,
+					Claims:   nil,
+				},
+				ResizePolicy: nil,
+				VolumeMounts: []apiv1.VolumeMount{
+					{
+						Name:             hostVolumeName,
+						ReadOnly:         false,
+						MountPath:        hostMountPath,
+						SubPath:          "",
+						MountPropagation: nil,
+						SubPathExpr:      "",
+					},
+				},
+				VolumeDevices:            nil,
+				LivenessProbe:            nil,
+				ReadinessProbe:           nil,
+				StartupProbe:             nil,
+				Lifecycle:                nil,
+				TerminationMessagePath:   "",
+				TerminationMessagePolicy: "",
+				ImagePullPolicy:          "",
+				SecurityContext: &apiv1.SecurityContext{
+					Privileged:               &isPrivileged,
+					Capabilities:             nil,
+					SeccompProfile:           nil,
+					ProcMount:                nil,
+					ReadOnlyRootFilesystem:   nil,
+					AllowPrivilegeEscalation: nil,
+					RunAsNonRoot:             nil,
+					RunAsGroup:               nil,
+					RunAsUser:                nil,
+					SELinuxOptions:           nil,
+					WindowsOptions:           nil,
+				},
+				Stdin:     false,
+				StdinOnce: false,
+				TTY:       false,
+			},
+		},
+		[]apiv1.Volume{
+			{
+				Name:         hostVolumeName,
+				VolumeSource: manager.GetVolumeSourceForHostPath("/"), // mount the entire host filesystem in this volume
+			},
+		},
+		"",
+		"",
+		nil,
+		map[string]string{
+			"kubernetes.io/hostname": nodeName,
```

Comment:
What's this?

Reply:
replaced `kubernetes.io/hostname` a variable from the k8s api library but it is the node selector used to specify the hostname of the node for which to schedule the pod, in this case we're using the hostname label to specify the pod.

i also turned that map into a variable called `nodeSelectors`

Reply:
https://kubernetes.io/docs/reference/labels-annotations-taints/#kubernetesiohostname